apollo-backend/src/test/java/io/logz/apollo/helpers/ModelsGenerator.java

```diff
@@ -144,121 +167,85 @@ public static Group createGroup() {
-    }
+    public static DeployableVersion createDeployableVersion(Service relatedService, String repositoryUrl, String commitSha) {
+        DeployableVersion testDeployableVersion = new DeployableVersion();
+        testDeployableVersion.setGitCommitSha(commitSha);
+        testDeployableVersion.setGithubRepositoryUrl(repositoryUrl);
+        testDeployableVersion.setServiceId(relatedService.getId());
+        testDeployableVersion.setCommitDate(Date.from(LocalDateTime.now(ZoneId.of("UTC")).atZone(ZoneId.systemDefault()).toInstant()));
 
-    public static Group createAndSubmitGroup(ApolloTestClient apolloTestClient, int environmentId) throws ApolloClientException {
-        return createAndSubmitGroup(apolloTestClient, createAndSubmitService(apolloTestClient).getId(), environmentId);
+        return testDeployableVersion;
     }
 
-    public static Group createAndSubmitGroup(ApolloTestClient apolloTestClient, int serviceId, int environmentId) throws ApolloClientException {
-        Group testGroup = createGroup();
-
-        testGroup.setServiceId(serviceId);
-        testGroup.setEnvironmentId(environmentId);
-
-        apolloTestClient.addGroup(testGroup);
-
-        testGroup.setId(apolloTestClient.getGroupByName(testGroup.getName()).getId());
+    public static DeployableVersion createDeployableVersion(Service relatedService) {
+        String repositoryUrl = "http://test.com/logzio/" + Common.randomStr(5);
+        String commitSha = "abc129aed837f6" + Common.randomStr(5);
 
-        return testGroup;
+        return createDeployableVersion(relatedService, repositoryUrl, commitSha);
     }
 
-    public static Group createAndSubmitGroup(ApolloTestClient apolloTestClient, int serviceId, int environmentId, String name) throws ApolloClientException {
-        Group testGroup = createGroup();
+    public static DeployableVersion createAndSubmitDeployableVersion(ApolloTestClient apolloTestClient, Service service, String repositoryUrl, String commitSha) throws ApolloClientException {
 
-        testGroup.setServiceId(serviceId);
-        testGroup.setEnvironmentId(environmentId);
-        testGroup.setName(name);
+        // Add deployable version
+        DeployableVersion testDeployableVersion = ModelsGenerator.createDeployableVersion(service, repositoryUrl, commitSha);
+        testDeployableVersion.setId(apolloTestClient.addDeployableVersion(testDeployableVersion).getId());
 
-        apolloTestClient.addGroup(testGroup);
+        return testDeployableVersion;
+    }
 
-        testGroup.setId(apolloTestClient.getGroupByName(testGroup.getName()).getId());
+    public static DeployableVersion createAndSubmitDeployableVersion(ApolloTestClient apolloTestClient, Service service) throws ApolloClientException {
+        String repositoryUrl = "http://test.com/logzio/" + Common.randomStr(5);
+        String commitSha = "abc129aed837f6" + Common.randomStr(5);
```

Comment:
Lets create `createAndSubmitDeployableVersion(apolloTestClient, service);` which reuse createDeployableVersion with the defaults so we will have only once place that generate URLs and Commit sha